LLK/scratch-www
# fix: remove Sentry and rewrite ErrorBoundary tests

### Resolves:

Half of ENA-224

### Changes:

We're no longer planning to use Sentry in this way, so this change removes all references to Sentry (the library) throughout scratch-www's code, README, and Webpack config. Note that Sentry (the organization) is still mentioned in the content of some pages.

Sentry support was tightly integrated with the ErrorBoundary component, which would report an error to Sentry and then display a crash message that includes the Sentry event ID. Now, ErrorBoundary skips sending the error to Sentry but still displays the crash message.

### Test Coverage:

The `ErrorBoundary` test sequence has been rewritten to verify that it switches from displaying children to displaying a crash message when it catches an error. Previously, the test verified that the component would call a mock Sentry interface, but did not verify user-facing behavior.

This is a little tricky to test manually. One way I've found:
1. run `npm start`
2. visit `http://localhost:8333//explore/projects/all`
3. open the dev console
4. search for `'masthead'` (with single quotes) in `explore.jsx`
5. click the `{}` button to pretty-print the file
6. find the `render()` function near `'masthead'` (for me, it's on line 197 of the pretty-printed file)
7. put a breakpoint on the `return React.createElement(` line (`React` might be a single-letter variable on minified builds)
8. reload the page so that the breakpoint hits
9. type `React=null` in the console (replace `React` with the appropriate single-letter variable if using a minified build)
10. press the "continue" button
